src/main/java/org/itsallcode/openfasttrace/report/PlainTextReport.java

```diff
@@ -219,36 +222,115 @@ private void renderMaturity(final LinkedSpecificationItem item, final PrintStrea
+
     private void renderAll(final PrintStream report)
     {
         this.trace.getItems().stream() //
                 .sorted(LINKED_ITEM_BY_ID) //
-                .forEachOrdered(item -> renderItemDetails(item, report));
+                .forEachOrdered(item -> renderItemDetails(report, item));
+    }
+
+    private void renderItemDetails(final PrintStream report, final LinkedSpecificationItem item)
+    {
+        renderItemSummary(report, item);
+        renderDescription(report, item);
+        renderLinks(report, item);
+        renderTags(report, item);
+        renderItemDetailsEnd(report);
     }
 
-    private void renderItemDetails(final LinkedSpecificationItem item, final PrintStream report)
+    private void renderEmptyItemDetailsLine(final PrintStream report)
     {
-        renderItemSummary(item, report);
+        report.print("|");
+        report.print(this.newline);
+    }
 
+    private void renderDescription(final PrintStream report, final LinkedSpecificationItem item)
+    {
         final String description = item.getDescription();
         if (description != null && !description.isEmpty())
         {
-            report.print("|");
-            report.print(this.newline);
+            renderEmptyItemDetailsLine(report);
             for (final String line : description.split(Newline.anyNewlineReqEx()))
             {
                 report.print("| ");
                 report.print(line);
                 report.print(this.newline);
             }
-            report.print("|");
-            report.print(this.newline);
+            ++this.nonEmptySections;
         }
+    }
+
+    // [impl->dsn~reporting.plain-text.link-details~1]
+    private void renderLinks(final PrintStream report, final LinkedSpecificationItem item)
+    {
+
+        final Map<LinkStatus, List<LinkedSpecificationItem>> links = item.getLinks();
+        if (!links.isEmpty())
+        {
+            renderEmptyItemDetailsLine(report);
+            renderOrderedLinks(report, links);
+            ++this.nonEmptySections;
+        }
+    }
+
+    private void renderOrderedLinks(final PrintStream report,
+            final Map<LinkStatus, List<LinkedSpecificationItem>> links)
+    {
+        final List<TracedLink> tracedLinks = flattenTracedLinks(links);
+        tracedLinks.stream() //
+                .sorted((a, b) -> a.getOtherLinkEnd().getId().compareTo(b.getOtherLinkEnd().getId())) //
+                .forEachOrdered(link -> renderLink(report, link));
+    }
+
+    private List<TracedLink> flattenTracedLinks(
+            final Map<LinkStatus, List<LinkedSpecificationItem>> links)
+    {
+        final List<TracedLink> tracedLinks = new ArrayList<>();
+        for (final Entry<LinkStatus, List<LinkedSpecificationItem>> entry : links.entrySet())
+        {
+            for (final LinkedSpecificationItem other : entry.getValue())
+            {
+                tracedLinks.add(new TracedLink(other, entry.getKey()));
+            }
+        }
```

Comment:
Maybe use a stream with `map` instead of nested `for` loop

Reply:
Tried that first but the code was harder to read ... and slower.